### PR TITLE
Fix admin node

### LINKED ISSUE #98: AdminNode add a new node display error when db has no node data
image

### PR DESCRIPTION
**Pull request by noneback**
fix : AdminNode add a new node display error when db has no node data

Fix: https://github.com/casbin/casbin-forum/issues/98

### CODE REVIEW

**Comment by noneback** [comment]
fix #98

File: web/src/TopicPage.js [hunk 1]
```diff
@@ -23,13 +23,17 @@ import {withRouter, Link} from "react-router-dom";
 import moment from "moment";
 import i18next from "i18next";
 
+
 class TopicPage extends React.Component {
   constructor(props) {
     super(props);
+    const rootTabId = "all" // should be set as the root tab id
```
Comment (hsluoyz) [inline web/src/TopicPage.js:30]: Don't commit this part again.

File: web/src/TopicPage.js [hunk 2]
```diff
@@ -23,13 +23,17 @@ import {withRouter, Link} from "react-router-dom";
 import moment from "moment";
 import i18next from "i18next";
 
+
```
Comment (hsluoyz) [inline web/src/TopicPage.js:26]: no new line

File: web/src/admin/AdminNode.js [hunk 3]
```diff
@@ -444,10 +444,17 @@ class AdminNode extends React.Component {
...
     switch (item) {
       case "node":
-        value = this.getIndexFromNodeId(this.state.form.parentNode);
-        data= this.state.nodes.map((node, i) => {
-          return {text: `${node.nodeInfo.name} / ${node.nodeInfo.id}`, id: i};
-        });
+        if (this.state.form.parentNode){
+          value = this.getIndexFromNodeId(this.state.form.parentNode);
+        }
+        if (this.state.nodes){
```
Comment (hsluoyz) [inline web/src/admin/AdminNode.js:450]: Necessary?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/admin/AdminNode.js (hunk 3 above)
Comment (noneback) [reply web/src/admin/AdminNode.js:450]: Actually on my local frontend test, if I remove them, still got the same error like #98.

Comment (hsluoyz) [reply web/src/admin/AdminNode.js:450]: I mean `if (this.state.nodes){`

**Comment by hsluoyz** [comment]
Clean your commits.

File: web/src/admin/AdminNode.js (hunk 3 above)
Comment (noneback) [reply web/src/admin/AdminNode.js:450]: If remove `if (this.state.nodes)` and no node in db,then:
err
The `this.state.nodes` is initialized to an array at the beginning,but somehow be set to null or undefined while page rendering(maybe in function `getNodes`)

**Comment by noneback** [comment]
Can I make a new pr to replace this one?

**Comment by hsluoyz** [comment]
Modify the existing PR.

**Comment by noneback** [comment]
I have no idea about how to clean the commits,because I delete my forked repository by mistake when I want to clean it.

**Comment by noneback** [comment]
Sorry.

File: web/src/admin/AdminNode.js (hunk 3 above)
Comment (hsluoyz) [reply web/src/admin/AdminNode.js:450]: getNodes() should return [] for empty. Please fix at getNodes() side.

**Comment by noneback** [comment]
replaced by #103

**Pull request closed without merging**